### ISSUE TITLE
Non unique experiment names

### ISSUE THREAD

**Issue by oyvindeide** [equinor/ert#8021]
While there is no actual requirement that experiment names needs to be unique, it can cause confusion if there are multiple experiments with the same name.

There are a few possible solutions:
1. Enforce unique experiment names, this would make it easier to differentiate, but makes it slightly harder for the user as they have to make up a new name
2. Enforce unique name, but append for example the date to the suggested name, still allowing the user to change if they want
3. Allow all names, but always differentiate by `date_created`, which is a kind of variant of 2., would also include showing the data (or some metadata) as hover info in the plotter
4. No enforcement
5. ?

**Comment by dafeda** [comment]
I think we should enforce unique experiment names because it makes it easier for the user to differentiate ensembles in the plotter and in the experiment manager.

**Comment by oyvindeide** [comment]
I agree that unique names are good, but got multiple users complaining that running `Ensemble experiment` was now more cumbersome, and they just ended up writing gibberish in the experiment name field. They requested either an overwrite functionality, or some kind auto name feature. Maybe we should just allow duplicate names for the moment, or add a (i) after the name if duplicate?

**Comment by dafeda** [comment]
I mean, if our users are happy working with duplicate names it's fine by me.

**Comment by oyvindeide** [comment]
After discussing we circled back to unique experiment names, but to avoid user frustration we will add `_%d` after the experiment name if it is duplicate, and if the name already has `_%d`, we will increment that number.

**Comment by oyvindeide** [comment]
There should also be a requirement in `LocalStorage`.
